display vulns in package.json

bmvermeer opened the pull request.

bmvermeer committed:
display vulns in package.json

bmvermeer committed:
display vulns in package.json

thisislawatts commented on src/getImports/packageJsonParser.js:
```diff
@@ -0,0 +1,37 @@
+
+  const pjson = JSON.parse(source);
+
+  for (var dep in pjson.dependencies) {
```
Should we also be scanning folks [devDependencies](https://docs.npmjs.com/files/package.json#devdependencies) and [peerDependencies](https://docs.npmjs.com/files/package.json#peerdependencies)?

bmvermeer commented:
Maybe in the future. I dont believe it is good to do this by default. I think it will be considered noise but lets monitor this. 
Maybe we should make it optional just like in the CLI.

thisislawatts commented on src/getImports/packageJsonParser.js:
```diff
@@ -0,0 +1,37 @@
+    p.fileName = fileName;
+    p.name = dep;
+    p.loc = findLoc(dep, lines);
+    p.line = p.loc.start.line;
```
I think we're missing the `version` property. Which looks is being returned by the `htmlParser`:
https://github.com/snyk/vulncost/blob/77dc018fb3282cbac05486389ef0a8855ffc643e/src/getImports/htmlParser.js#L91

We should also aim to validate the version string with `semver.valid`, to filter out instances where users are directly referencing GitHub repos (although we could also test these 🤔).

Here's a gist based on the npm documentation for permitted dep declaration styles, filtering out those which can't be resolved to a version.
https://gist.github.com/thisislawatts/48e5c6bdbdaa4156e18bd81f60380a06

Ideally we should use either the yarn.lock/package-lock.json which gives a more accurate idea of what will be installed during deployment.

bmvermeer replied:
Check the js / ts parser. By default we take the latest version if no manufest is available or else we check the version in the manifest.

For html when using a cdn we do not depend on a manifest file but the version inline.

thisislawatts commented:
@bmvermeer I am not sure I agree, given the user is in the manifest file and seeing Snyk annotations on their `dependencies` the absence of annotations on the `devDependencies` does not clearly communicate these are being ignored by default. Instead my assumption would be that these devDeps should be considered safe/secure.

thisislawatts committed:
test: adds tests for html parser

I would like to extend the html parser to support a new CDN but worried
about introducing a regression I am adding some tests to validate the current behaviour

thisislawatts committed:
fix: adds support for assets hosted on github

thisislawatts committed:
feat: adds support for ajax.aspnet.com

thisislawatts committed:
test: adds CDNjs scenario

thisislawatts committed:
feat: adds Command to remove auth token from local config

It should be possible to remove your API token from the IDE

thisislawatts committed:
fix: do not scan local modules

Do not treat local modules as if they are 3rd party
packages hosted on npm.

thisislawatts committed:
fix: supports eg/file.js

The NodeJS require behaviour will also load a file if it's
exists in the file directory

thisislawatts committed:
feat: adds setting to ignore based on paths

Introduces a configuration option to ignore a path based on a regex.

thisislawatts commented:
@bmvermeer In case my earlier edit got missed we should definitely filter out any packages which are not being loaded from NPM, this includes the GitHub & local style mentioned here:
https://docs.npmjs.com/files/package.json#github-urls

thisislawatts committed:
feat: use js|typescript config regex

Reuse JS/Typescript configuration options
to validate the found files rather than hard coding
file extensions.

bmvermeer committed:
Merge branch 'master' into decorate-package.json

bmvermeer committed:
Merge branch 'master' into feat/adds-command-to-log-out-of-editor

bmvermeer committed:
Merge pull request #15 from snyk/feat/adds-command-to-log-out-of-editor

feat: adds command to remove auth token from local config

thisislawatts committed:
docs: update README/changelogs to detail supported CDNs

thisislawatts committed:
Merge pull request #14 from snyk/test/add-tests-for-html-parser

feat: adds support for the Microsoft Ajax Content Delivery Network

thisislawatts committed:
chore: format

bmvermeer committed:
Update CHANGELOG.md

update CHANGELOG

bmvermeer committed:
Merge pull request #16 from snyk/fix/do-not-scan-local-modules

fix: do not scan local modules

bmvermeer committed:
rebase with master

bmvermeer committed:
Merge branch 'decorate-package.json' of https://github.com/snyk/vulncost into decorate-package.json

bmvermeer committed:
entry in CHANGELOG

thisislawatts commented:
:tada: This PR is included in version 1.2.0 :tada:

The release is available on [GitHub release](https://github.com/snyk/vulncost/releases/tag/v1.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
